app/code/Magento/PageCache/Model/Varnish/VclGenerator.php

```diff
@@ -67,7 +74,8 @@ public function __construct(
         $accessList,
         $gracePeriod,
         $sslOffloadedHeader,
-        $designExceptions = []
+        $designExceptions = [],
+        DirectoryList $directoryList = null
```

Comment:
Hi you also need to add the DirectoryList as optional ?DirectoryList $directoryList

Reply:
@rodrigowebjump This isn't mentioned in DevDocs (https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php), so im not sure about that.
```
    public function __construct(
        \Old\Dependency\Interface $oldDependency,
        $oldRequiredConstructorParameter,
        $oldOptionalConstructorParameter = null,
        \New\Dependency\Interface $newDependency = null
    ) {
        ...
        $this->newDependency = $newDependency ?: \Magento\Framework\App\ObjectManager::getInstance()->get(\New\Dependency\Interface::class);
    }
```

Reply:
Hi @torhoehn 

This is a good practice.
I create a PR for this https://github.com/magento/devdocs/pull/6033

It's is supported since PHP 7.1 (https://www.php.net/manual/en/migration71.new-features.php) and is recommended when adding a nullable param.

Reply:
@rodrigowebjump Okay, I added it. :)